src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/AlleleSubsettingUtils.java

```diff
@@ -0,0 +1,233 @@
+
+            final int[] alleleCounts = new GenotypeLikelihoodCalculators()
+                    .getInstance(ploidy, vc.getNAlleles()).genotypeAlleleCountsAt(indexOfMostLikelyGenotype)
+                    .alleleCountsByIndex(vc.getNAlleles() - 1);
+
+            for (int allele = 1; allele < alleleCounts.length; allele++) {
+                likelihoodSums[allele] += alleleCounts[allele] * GLDiffBetweenRefAndBest;
+            }
+        }
+
+        final List<Double> properAltAlleleLikelihoodSums = Arrays.stream(properAltAlleleIndices)
+                .mapToObj(n -> likelihoodSums[n]).collect(Collectors.toList());
+        Collections.sort(properAltAlleleLikelihoodSums, Collections.reverseOrder());
+        final double likelihoodSumThreshold = properAltAlleleLikelihoodSums.get(numAltAllelesToKeep);
+        return IntStream.range(0, vc.getNAlleles()) //keep ref, non-ref, and alts above the threshold
+                .filter(n -> n == 0 || n == nonRefAltAlleleIndex || likelihoodSums[n] > likelihoodSumThreshold)
+                .mapToObj(n -> vc.getAlternateAllele(n-1)).collect(Collectors.toList());
+    }
+
+    /**
+     * Given a list of original alleles and a subset of new alleles to retain, find the array of old PL indices that correspond
+     * to new PL indices i.e. result[7] = old PL index of genotype containing same alleles as the new genotype with PL index 7.
+     *
+     * This method is written in terms f indices rather than subsetting PLs directly in order to produce output that can be
+     * recycled from sample to sample, provided that the ploidy is the same.
+     *
+     * @param ploidy                Ploidy (number of chromosomes describing PL's)
+     * @param originalAlleles       List of original alleles
+     * @param newAlleles            New alleles -- must be a subset of {@code originalAlleles}
+     * @return                      old PL indices of new genotypes
+     */
+    private static int[] subsettedGenotypeLikelihoodIndices(final int ploidy, final List<Allele> originalAlleles, final List<Allele> newAlleles) {
+        final int[] result = new int[GenotypeLikelihoods.numLikelihoods(newAlleles.size(), ploidy)];
+        final List<Boolean> isAlleleKept = keptAlleleIndices(originalAlleles, newAlleles);
+        final Permutation<Allele> allelePermutation = new IndexedAlleleList<>(originalAlleles).permutation(new IndexedAlleleList<>(newAlleles));
+
+        final GenotypeLikelihoodCalculator glCalc = GL_CALCS.getInstance(ploidy, originalAlleles.size());
+        for (int oldPLIndex = 0; oldPLIndex < glCalc.genotypeCount(); oldPLIndex++) {
+            final GenotypeAlleleCounts oldAlleleCounts = glCalc.genotypeAlleleCountsAt(oldPLIndex);
+
+            final boolean onlyNewAlleles = IntStream.range(0, oldAlleleCounts.distinctAlleleCount())
+                    .map(oldAlleleCounts::alleleIndexAt).allMatch(isAlleleKept::get);
```

Comment:
I got the impression that some of the operations done here could be moved into the ```Permutation``` heriarchy. For example what about adding a method in ```Permutation``` to check whether an original index is kept?:

```java
public interface Permutation<A> {
  // ...
  boolean isKept(final int fromIndex);
  boolean isKept(final A from);
 
}
```